Downgrade .Net Full framework target to net451

smadala opened the pull request:
Default .Net framework on Windows Server 2012 R2 is net451([doc](https://blogs.msdn.microsoft.com/astebner/2007/03/14/mailbag-what-version-of-the-net-framework-is-included-in-what-version-of-the-os/)).  This PR enable running vstest.console.exe directly on windows server 2012 without any additional installation.

smadala committed:
Replace net46 with net451

smadala committed:
keep external csproj target framework as net46

smadala committed:
Merge branch 'master' into retarget-to-net451-03

smadala committed:
Fix Acceptance tests

codito commented:
Require validation on Mono 5.0.1.1. Ensure we don't regress the behavior.

codito commented on scripts/build.sh:
```diff
@@ -87,7 +87,7 @@ DOTNET_RUNTIME_VERSION="2.0.0-preview2-25331-01"
 #
 TPB_Solution="TestPlatform.sln"
-TPB_TargetFramework="net46"
+TPB_TargetFramework="net451"
```
Ensure this script works in mono.

smadala replied:
Tested it. worked fine.

codito commented on src/Microsoft.TestPlatform.PlatformAbstractions/common/System/ProcessHelper.cs:
```diff
@@ -35,7 +35,11 @@ public object LaunchProcess(string processPath, string arguments, string working
                 {
                     foreach (var kvp in envVariables)
                     {
+#if NET451
```
Consider using EnvironmentVariables always.

smadala replied:
API not available for netcoreapp1.0.

codito replied:
Suggest to move this out.

codito commented on src/datacollector/datacollector.csproj:
```diff
@@ -9,13 +9,13 @@
     <WarningsAsErrors>true</WarningsAsErrors>
-    <PackageTargetFallback Condition=" '$(TargetFramework)' != 'net46' ">$(PackageTargetFallback);dnxcore50;netstandardapp1.0;portable-net45+win8;portable-net45+wp80+win8+wpa81+dnxcore50</PackageTargetFallback>
-    <PlatformTarget Condition="'$(TargetFramework)' == 'net46'">AnyCPU</PlatformTarget>
+    <PackageTargetFallback Condition=" '$(TargetFramework)' != 'net451' ">$(PackageTargetFallback);dnxcore50;netstandardapp1.0;portable-net45+win8;portable-net45+wp80+win8+wpa81+dnxcore50</PackageTargetFallback>
```
Consider removing packagetargetfallback

smadala replied:
Done.

codito commented on src/testhost/testhost.csproj:
```diff
@@ -6,11 +6,11 @@
+    <TargetFrameworks>netcoreapp1.0;net451</TargetFrameworks>
     <OutputType>Exe</OutputType>
-    <PackageTargetFallback Condition=" '$(TargetFramework)' != 'net46' ">$(PackageTargetFallback);dnxcore50;netstandardapp1.0;portable-net45+win8;portable-net45+wp80+win8+wpa81+dnxcore50</PackageTargetFallback>
+    <PackageTargetFallback Condition=" '$(TargetFramework)' != 'net451' ">$(PackageTargetFallback);dnxcore50;netstandardapp1.0;portable-net45+win8;portable-net45+wp80+win8+wpa81+dnxcore50</PackageTargetFallback>
```
Consider removing this

codito commented on test/Microsoft.TestPlatform.ObjectModel.PlatformTests/project.json:
```diff
@@ -41,7 +41,7 @@
       }
     },
-    "net46": {}
+    "net451": {}
```
Consider removing the project.json

smadala replied:
Done.

codito commented on test/Microsoft.TestPlatform.ObjectModel.UnitTests/FrameworkTests.cs:
```diff
@@ -23,15 +23,15 @@ public void FrameworkFromStringShouldTrimSpacesAroundFrameworkString()
         {
-#if NET46
+#if NET451
             Assert.AreEqual(".NETFramework,Version=v4.6", Framework.DefaultFramework.Name);
```
Default framework continues to be `4.6`?

smadala replied:
Made 451 as default, To avoid confusion.

codito commented on test/Microsoft.TestPlatform.PerformanceTests/project.json:
```diff
@@ -41,7 +41,7 @@
       }
     },
-    "net46": { }
+    "net451": { }
```
Consider removing project.json

codito commented on test/Microsoft.TestPlatform.SmokeTests/DataCollectorTests.cs:
```diff
@@ -39,7 +39,17 @@ private static void ValidateInProcDataCollectionOutput()
-            var inprocasm = testEnvironment.GetTestAsset("SimpleDataCollector.dll");
+            string inprocasm = null;
+
+            // SimpleDataCollector has dependency on TestPlatform OM, which targets net46. This can be removed after this PR merge.
```
Isn't this a project 2 project reference?

smadala replied:
Done.

codito commented on test/Microsoft.TestPlatform.TestUtilities/FileUtility.cs:
```diff
@@ -0,0 +1,24 @@
+    /// <summary>
+    /// Assembly utility to perform assembly related functions.
+    /// </summary>
+    public class FileUtility
```
Shouldn't this be available in the test base class?

smadala replied:
Removed this.

codito commented on test/Microsoft.TestPlatform.TestUtilities/IntegrationTestEnvironment.cs:
```diff
@@ -190,15 +190,15 @@ public string RunnerFramework
         /// produce <c>TestAssets\SimpleUnitTest\bin\Debug\SimpleUnitTest.dll</c>
         /// </remarks>
-        public string GetTestAsset(string assetName)
+        public string GetTestAsset(string assetName, string targetFramework = null)
```
Not sure why this is required.

smadala replied:
To run xunit acceptance test which won't support net 451. Added overload method.

codito commented on test/TestAssets/OutOfProcDataCollector/SampleDataCollector.cs:
```diff
@@ -43,14 +44,14 @@ private void Events_TestCaseStart(object sender, TestCaseStartEventArgs e)
             this.logger.LogWarning(this.context.SessionDataCollectionContext, "TestCaseStarted " + e.TestCaseName);
             this.logger.LogWarning(this.context.SessionDataCollectionContext, "TestCaseStarted " + e.TestElement.FullyQualifiedName);
-            var filename = Path.Combine(AppContext.BaseDirectory, "testcasefilename" + i++ + ".txt");
+            var filename = Path.Combine(FileUtility.GetAppDomainBaseDir(), "testcasefilename" + i++ + ".txt");
```
A test asset is now dependent on a test utility. It will pull all unnecessary dependencies from the product/test code.

smadala replied:
Removed this.

harshjain2 commented:
@smadala Ensure this change doesn't degrade performance.

smadala committed:
Remove PackageTargetFallback property, project.json
Make full framework default target version to .NETFramework,Version=v4.5.1
Make simple data collector refence tp packages by project reference.

smadala committed:
* Use temp directory rather than app domain base directory
* Don't try to run tests for all frameworks

smadala committed:
Close/Dispose tcp client

smadala committed:
Merge branch 'master' into retarget-to-net451-03

smadala committed:
Merge with blame data collector

smadala committed:
Make TP packages project reference in BlameUnitTestProject

smadala committed:
Merge branch 'master' into retarget-to-net451-03

smadala committed:
Merge branch 'master' of https://github.com/Microsoft/vstest into retarget-to-net451-03

smadala committed:
Merge branch 'retarget-to-net451-03' of https://github.com/smadala/vstest into retarget-to-net451-03

smadala commented:
@harshjain2 Blow were the perf numbers, Looks like NET46 has slight edge. 
![net451_vs_net46_perf](https://user-images.githubusercontent.com/4337332/28259440-e29f15c6-6af3-11e7-9b68-cad7c3dc7358.png)

smadala commented:
@dotnet-bot Please test Windows_NT / Debug Build

smadala commented:
@dotnet-bot Test Windows_NT / Debug Build please.

harshjain2 commented:
@smadala Perf numbers looks good to me...

Faizan2304 commented on scripts/build/TestAssets.props:
```diff
@@ -3,7 +3,6 @@
     <PropertyGroup>
         <!-- Not able to add TargetFrameworks here due to: https://github.com/dotnet/sdk/issues/1110 -->
         <NetCoreAppTargetFrameWork Condition="'$(TargetFramework)' != '' AND $(TargetFramework.StartsWith('netcoreapp'))">true</NetCoreAppTargetFrameWork>
-	    <PackageTargetFallback Condition="$(NetCoreAppTargetFrameWork) == 'true' ">$(PackageTargetFallback);dnxcore50;portable-net45+win8</PackageTargetFallback>
```
Why don't we need this now?

smadala replied:
It appears all of our dependent packages have built for netstandard1.5 and netcoreapp1.0. We no more need `PackageTargetFallback`, unless we add a new dependency which didn't build for netstandard1.5 or netcoreapp1.0.

codito commented on test/Microsoft.TestPlatform.Build.UnitTests/Microsoft.TestPlatform.Build.UnitTests.csproj:
```diff
@@ -27,7 +26,7 @@
     </PackageReference>
   </ItemGroup>
-  <ItemGroup Condition=" '$(TargetFramework)' == 'net46' ">
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net451' ">
```
nit: remove

codito commented on test/TestAssets/BlameUnitTestProject/BlameUnitTestProject.csproj:
```diff
@@ -1,31 +1,26 @@
-    <!--PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.3.0-preview-20170628-02" />
-    <PackageReference Include="MSTest.TestAdapter" Version="1.1.18" />
-    <PackageReference Include="MSTest.TestFramework" Version="1.1.18" /-->
+  <PropertyGroup Condition=" '$(Configuration)' == 'Release' ">
```
May not be required

smadala committed:
- Add Extension method to add Environment Variable for ProcessStartInfo
- Fix Nit comments.
